crates/builder/src/tasks/submit.rs

```diff
@@ -218,8 +250,33 @@ impl SubmitTask {
         let handle = tokio::spawn(async move {
             loop {
                 if let Some(in_progress) = inbound.recv().await {
-                    if let Err(e) = self.handle_inbound(&in_progress).await {
-                        error!(%e, "error in block submission. Dropping block.");
+                    let mut retries = 0;
+                    loop {
+                        match self.handle_inbound(&in_progress).await {
+                            Ok(ControlFlow::Retry) => {
+                                retries += 1;
+                                if retries > 3 {
+                                    tracing::error!(
+                                        "error handling inbound block: too many retries"
+                                    );
+                                    break;
+                                }
+                                tracing::error!("error handling inbound block: retrying");
+                                tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+                            }
+                            Ok(ControlFlow::Skip) => {
+                                tracing::info!("skipping block");
+                                break;
+                            }
+                            Ok(ControlFlow::Done) => {
+                                tracing::info!("block landed successfully");
```

Comment:
need to set retries to 0 in Skip and Done arms

Reply:
pretty sure it gets reset on each iteration of the outer loop?

Reply:
oh you're right. i don't like the nested loop tho. kinda ugly and implies we should break out a function

Reply:
yep agreed, we can break it out in another PR